diff --git a/resources/node_resource.inc b/resources/node_resource.inc
index 8f870b7..6669a1a 100644
--- a/resources/node_resource.inc
+++ b/resources/node_resource.inc
@@ -339,8 +339,9 @@ function _node_resource_create($node) {
   );
   $stub_form = drupal_build_form($node_type . '_node_form', $stub_form_state);
   $form_state['triggering_element'] = $stub_form['actions']['submit'];
+  $node = (object) array_merge((array) $stub_node, (array) $node);
 
-  drupal_form_submit($node_type . '_node_form', $form_state, (object)$stub_node);
+  drupal_form_submit($node_type . '_node_form', $form_state, $node);
 
   if ($errors = form_get_errors()) {
     return services_error(implode(" ", $errors), 406, array('form_errors' => $errors));
@@ -423,6 +424,7 @@ function _node_resource_update($nid, $node) {
 
   $node_type = $node['type'];
   node_object_prepare($old_node);
+  $old_node = (object) array_merge((array) $old_node, (array) $node);
 
   // Setup form_state.
   $form_state = array();
diff --git a/resources/user_resource.inc b/resources/user_resource.inc
index 304a293..c1f9b5a 100644
--- a/resources/user_resource.inc
+++ b/resources/user_resource.inc
@@ -471,6 +471,7 @@ function _user_resource_update($uid, $account) {
     return services_error(t('You are not allowed to change your username.'), 406);
   }
 
+  $account_loaded = (object) array_merge((array) $account_loaded, (array) $account);
   $form_state['values']['op'] = variable_get('services_user_save_button_resource_update', t('Save'));
   $form_state['values']['#user_category'] = $category;
   $form_state['values']['#account'] = $account_loaded;
