diff --git a/docroot/profiles/rs_beliana/modules/contrib/views/plugins/views_plugin_style.inc b/docroot/profiles/rs_beliana/modules/contrib/views/plugins/views_plugin_style.inc
index fe12607..f854f39 100644
--- a/docroot/profiles/rs_beliana/modules/contrib/views/plugins/views_plugin_style.inc
+++ b/docroot/profiles/rs_beliana/modules/contrib/views/plugins/views_plugin_style.inc
@@ -532,7 +532,14 @@ class views_plugin_style extends views_plugin {
         foreach ($result as $count => $row) {
           $this->view->row_index = $count;
           foreach ($keys as $id) {
-            $this->rendered_fields[$count][$id] = $this->view->field[$id]->theme($row);
+            //MS $this->rendered_fields[$count][$id] = $this->view->field[$id]->theme($row);
+	    //MS start
+	    if ($id == "field_text_hesla"){
+            	$this->rendered_fields[$count][$id] = $row->_field_data["nid"]["entity"]->field_text_hesla["und"][0]["value"];
+	    } else {
+            	$this->rendered_fields[$count][$id] = $this->view->field[$id]->theme($row);
+	    } 
+	    // MS end
           }
 
           $this->row_tokens[$count] = $this->view->field[$id]->get_render_tokens(array());
