diff --git a/docroot/profiles/rs_beliana/modules/contrib/ckeditor/ckeditor.ckeditor.inc b/docroot/profiles/rs_beliana/modules/contrib/ckeditor/ckeditor.ckeditor.inc
index a7bc9f8..1ecd807 100644
--- a/docroot/profiles/rs_beliana/modules/contrib/ckeditor/ckeditor.ckeditor.inc
+++ b/docroot/profiles/rs_beliana/modules/contrib/ckeditor/ckeditor.ckeditor.inc
@@ -230,5 +230,36 @@ function ckeditor_ckeditor_plugin() {
     }
   }
 
+ // Set icons for the LITE plugin
+  // In LITE icons are not added in  plugin.js, thus the code above fails to add the correctly
+  if (isset($plugins['lite'])) {
+    $plugins['lite']['default'] = 't';
+    $plugins['lite']['buttons'] = array();
+    $plugins['lite']['buttons']['lite-acceptall']=array(
+        'label' => 'Accept all changes',
+        'icon' => 'icons/lite-acceptall.png',
+    );
+    $plugins['lite']['buttons']['lite-rejectall']=array(
+        'label' => 'Reject all changes',
+        'icon' => 'icons/lite-rejectall.png',
+    );
+    $plugins['lite']['buttons']['lite-acceptone']=array(
+        'label' => t('Accept change'),
+        'icon' => 'icons/lite-acceptone.png',
+    );
+    $plugins['lite']['buttons']['lite-rejectone']=array(
+        'label' => t('Reject change'),
+        'icon' => 'icons/lite-rejectone.png',
+    );
+    $plugins['lite']['buttons']['lite-toggleshow']=array(
+        'label' => t('Show/hide tracked changes'),
+        'icon' => 'icons/lite-toggleshow.png',
+    );
+    $plugins['lite']['buttons']['lite-toggletracking']=array(
+        'label' => t('Start/stop tracking changes'),
+        'icon' => 'icons/lite-toggletracking.png',
+    );
+  }
+
   return $plugins;
 }
