diff --git a/docroot/profiles/rs_beliana/modules/contrib/views_data_export/images/doc.png b/docroot/profiles/rs_beliana/modules/contrib/views_data_export/images/doc.png
index f12c9ef..0d3d3e0 100644
Binary files a/docroot/profiles/rs_beliana/modules/contrib/views_data_export/images/doc.png and b/docroot/profiles/rs_beliana/modules/contrib/views_data_export/images/doc.png differ
diff --git a/docroot/profiles/rs_beliana/modules/contrib/views_data_export/theme/views-data-export-doc-footer.tpl.php b/docroot/profiles/rs_beliana/modules/contrib/views_data_export/theme/views-data-export-doc-footer.tpl.php
index d75c23d..1fd5f4f 100644
--- a/docroot/profiles/rs_beliana/modules/contrib/views_data_export/theme/views-data-export-doc-footer.tpl.php
+++ b/docroot/profiles/rs_beliana/modules/contrib/views_data_export/theme/views-data-export-doc-footer.tpl.php
@@ -1,4 +1,2 @@
-      </tbody>
-    </table>
   </body>
 </html>
\ No newline at end of file
diff --git a/docroot/profiles/rs_beliana/modules/contrib/views_data_export/theme/views-data-export-doc-header.tpl.php b/docroot/profiles/rs_beliana/modules/contrib/views_data_export/theme/views-data-export-doc-header.tpl.php
index 86b824c..1461651 100644
--- a/docroot/profiles/rs_beliana/modules/contrib/views_data_export/theme/views-data-export-doc-header.tpl.php
+++ b/docroot/profiles/rs_beliana/modules/contrib/views_data_export/theme/views-data-export-doc-header.tpl.php
@@ -1,8 +1,16 @@
 <html>
   <head>
     <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <script type="text/x-mathjax-config">
+        MathJax.Hub.Config({
+          extensions: ['tex2jax.js'],
+          jax: ['input/TeX','output/HTML-CSS'],
+          tex2jax: {
+            inlineMath: [ ['$','$'], ['\\(','\\)'] ],
+            processEscapes: true
+          }
+        });
+    </script>
+    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
   </head>
   <body>
-    <table>
-    <?php print $header_row; ?>
-    <tbody>
\ No newline at end of file
diff --git a/docroot/profiles/rs_beliana/modules/contrib/views_data_export/theme/views_data_export.theme.inc b/docroot/profiles/rs_beliana/modules/contrib/views_data_export/theme/views_data_export.theme.inc
index 4e31fb0..9a18d93 100644
--- a/docroot/profiles/rs_beliana/modules/contrib/views_data_export/theme/views_data_export.theme.inc
+++ b/docroot/profiles/rs_beliana/modules/contrib/views_data_export/theme/views_data_export.theme.inc
@@ -302,8 +302,6 @@ function template_process_views_data_export_msoffice_body(&$vars) {
 
   // Construct the tbody of a table, see theme_table().
 
-  $ts = tablesort_init($vars['header']);
-
   $flip = array(
     'even' => 'odd',
     'odd' => 'even',
@@ -337,18 +335,29 @@ function template_process_views_data_export_msoffice_body(&$vars) {
       }
 
       // Build row
-      $output .= ' <tr' . drupal_attributes($attributes) . '>';
-      $i = 0;
       foreach ($cells as $cell) {
-        $cell = tablesort_cell($cell, $vars['header'], $ts, $i++);
-        $output .= _theme_table_cell($cell);
+        $output .= render($cell);
       }
-      $output .= " </tr>\n";
     }
   }
 
-
+  // Export na tlac
+  if ($vars['view']->name === 'export' && $vars['view']->current_display === 'views_data_export_1') {
+    $output = preg_replace('/<\/?(a|span) ?.*?>/', '', $output); // strip some html tags
+    $output = preg_replace('/➔ /', '→ ', $output); 
+    $output = preg_replace('/➔/', '→ ', $output); 
+    $vars['tbody'] = $output;
+  }
+  // Export pre sadzac
+  else if ($vars['view']->name === 'export_sadzac' && $vars['view']->current_display === 'views_data_export_1') {
+    $output = preg_replace('/<\/?(a|span) ?.*?>/', '', $output); // strip some other html tags
+    $output = preg_replace('/➔ /', '→ ', $output); 
+    $output = preg_replace('/➔/', '→ ', $output); 
+    $vars['tbody'] = $output;
+  }
+  else {
   $vars['tbody'] = preg_replace('/<\/?(a|span) ?.*?>/', '', $output); // strip 'a' and 'span' tags
+  }
 
 }
 
