diff --git a/docroot/profiles/rs_beliana/modules/contrib/link/link.module b/docroot/profiles/rs_beliana/modules/contrib/link/link.module
index 1e9d088..032fdda 100644
--- a/docroot/profiles/rs_beliana/modules/contrib/link/link.module
+++ b/docroot/profiles/rs_beliana/modules/contrib/link/link.module
@@ -1345,7 +1345,9 @@ function link_url_type($text) {
   $internal_pattern = "/^(?:[a-z0-9" . $LINK_ICHARS . "_\-+\[\] ]+)";
   $internal_pattern_file = "/^(?:[a-z0-9" . $LINK_ICHARS . "_\-+\[\]\. \/\(\)][a-z0-9" . $LINK_ICHARS . "_\-+\[\]\. \(\)][a-z0-9" . $LINK_ICHARS . "_\-+\[\]\. \/\(\)]+)$/i";
 
-  $directories = "(?:\/[a-z0-9" . $LINK_ICHARS . "_\-\.~+%=&,$'#!():;*@\[\]]*)*";
+  //MS: allow the '|' character in the directory part; needed by the freemap.sk links
+  #$directories = "(?:\/[a-z0-9" . $LINK_ICHARS . "_\-\.~+%=&,$'#!():;*@\[\]]*)*";
+  $directories = "(?:\/[a-z0-9" . $LINK_ICHARS . "\|_\-\.~+%=&,$'#!():;*@\[\]]*)*";
   // Yes, four backslashes == a single backslash.
   $query = "(?:\/?\?([?a-z0-9" . $LINK_ICHARS . "+_|\-\.~\/\\\\%=&,$'!():;*@\[\]{} ]*))";
   $anchor = "(?:#[a-z0-9" . $LINK_ICHARS . "_\-\.~+%=&,$'():;*@\[\]\/\?]*)";
