diff --git a/docroot/profiles/rs_beliana/modules/contrib/shs/includes/handlers/shs_handler_filter_term_node_tid_depth.inc b/docroot/profiles/rs_beliana/modules/contrib/shs/includes/handlers/shs_handler_filter_term_node_tid_depth.inc
index 99156e7..9618a76 100644
--- a/docroot/profiles/rs_beliana/modules/contrib/shs/includes/handlers/shs_handler_filter_term_node_tid_depth.inc
+++ b/docroot/profiles/rs_beliana/modules/contrib/shs/includes/handlers/shs_handler_filter_term_node_tid_depth.inc
@@ -90,6 +90,7 @@ class shs_handler_filter_term_node_tid_depth extends shs_handler_filter_term_nod
     if (module_exists('taxonomy_entity_index')) {
       $subquery = db_select('taxonomy_entity_index', 'tn');
       $subquery->addField('tn', 'entity_id');
+      $subquery->condition('entity_type', 'node');
     }
     else {
       $subquery = db_select('taxonomy_index', 'tn');
