Subject: [PATCH] BEL-132 - added module workbench_access
---
Index: docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.module
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.module b/docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.module
--- a/docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.module	(revision 52280f6c255e197feccd88cf99b166689eb103bc)
+++ b/docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.module	(date 1705558078324)
@@ -456,6 +456,12 @@
       }
     }
   }
+
+  foreach (module_implements('workbench_access_check') as $module) {
+    if(module_invoke($module, 'workbench_access_check', $op, $type, $access_ids, $account_access)){
+      return TRUE;
+    }
+  }
   return FALSE;
 }

Index: docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.views.inc
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.views.inc b/docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.views.inc
--- a/docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.views.inc	(revision 52280f6c255e197feccd88cf99b166689eb103bc)
+++ b/docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.views.inc	(date 1705557984698)
@@ -172,7 +172,7 @@
     'expose' => array(
       'label' => 'Sections',
       'identifier' => 'access_id',
-      'remember' => 1,
+      'remember' => 0,
       'single' => 0,
     ),
     'access_id' => array(),
Index: docroot/profiles/rs_beliana/modules/contrib/workbench_access/modules/taxonomy.workbench_access.inc
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/docroot/profiles/rs_beliana/modules/contrib/workbench_access/modules/taxonomy.workbench_access.inc b/docroot/profiles/rs_beliana/modules/contrib/workbench_access/modules/taxonomy.workbench_access.inc
--- a/docroot/profiles/rs_beliana/modules/contrib/workbench_access/modules/taxonomy.workbench_access.inc	(revision 52280f6c255e197feccd88cf99b166689eb103bc)
+++ b/docroot/profiles/rs_beliana/modules/contrib/workbench_access/modules/taxonomy.workbench_access.inc	(date 1705557884739)
@@ -286,7 +286,11 @@
  */
 function workbench_access_taxonomy_autocomplete_validate($element, &$form_state) {
   global $user;
-  $item = $form_state['values'][$element['#field_name']][$form_state['values']['language']];
+  $field = $form_state['values'][$element['#field_name']];
+  if(!isset($field[$form_state['values']['language']])){
+    return;
+  }
+  $item = $field[$form_state['values']['language']];
   if (empty($item)) {
     return;
   }
