diff --git a/docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.module b/docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.module
index de08d783..a6fcc43f 100644
--- a/docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.module
+++ b/docroot/profiles/rs_beliana/modules/contrib/workbench_access/workbench_access.module
@@ -555,16 +555,6 @@ function workbench_access_get_active_tree() {
 
   workbench_access_load_include($scheme);
 
-  $access_tree = &drupal_static(__FUNCTION__);
-  if (!isset($access_tree)) {
-    // Now check the cache.
-    $cache = cache_get('workbench_access_tree', 'cache_bootstrap');
-    if (isset($cache->data)) {
-      $access_tree = $cache->data;
-    }
-    if (isset($access_tree['access_scheme'])) {
-      return $access_tree;
-    }
     // Retrieve and cache data.
     $func = $scheme . '_workbench_access_info';
     $info = $func();
@@ -584,8 +574,7 @@ function workbench_access_get_active_tree() {
       'tree' => $tree,
       'active' => $active,
     );
-    cache_set('workbench_access_tree', $access_tree, 'cache_bootstrap');
-  }
+
   return $access_tree;
 }
 
